js/src/redux/providers/requestsActions.js

```diff
@@ -0,0 +1,257 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import BigNumber from 'bignumber.js';
+import store from 'store';
+
+import { ERROR_CODES } from '~/api/transport/error';
+import { trackRequest as trackRequestUtil, parseTransactionReceipt } from '~/util/tx';
+
+const LS_REQUESTS_KEY = '_parity::requests';
+
+class CachedRequests {
+  load (api) {
```

Comment:
I really would have preferred if these actions were tested, i.e. the full class. (It will become a pivotal piece, so rather get the base right and build up from there)